File: android/kotlin-davinci/davinci/src/main/java/com/pingidentity/samples/app/davinci/collector/Text.kt
/*
 * Copyright (c) 2024 - 2025 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package com.pingidentity.samples.app.davinci.collector

import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.wrapContentWidth
import androidx.compose.material3.OutlinedTextField
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.pingidentity.davinci.collector.TextCollector

@Composable
fun Text(
    field: TextCollector,
    onNodeUpdated: () -> Unit,
) {

    var isValid by remember(field) {
        mutableStateOf(true)
    }
    var text by remember(field) { mutableStateOf(field.value) }

    Row(
        modifier =
        Modifier
            .padding(4.dp)
            .fillMaxWidth(),
    ) {

        Spacer(modifier = Modifier.weight(1f, true))

        OutlinedTextField(
            modifier = Modifier.wrapContentWidth(Alignment.CenterHorizontally),
            value = text ,
            onValueChange = { value ->
                text = value
                field.value = value
                isValid = field.validate().isEmpty()
                onNodeUpdated()
            },
            isError = !isValid,
            supportingText = if (!isValid) {
                @Composable {
                    ErrorMessage(field.validate())
                }
            } else null,
            label = {
                androidx.compose.material3.Text(
                text = if (field.required) "${field.label}*" else field.label
            ) }
        )
        Spacer(modifier = Modifier.weight(1f, true))
    }
}

File: javascript/reactjs-todo-davinci/client/components/davinci-client/text.js
/*
 * ping-sample-web-react-davinci
 *
 * text.js
 *
 * Copyright (c) 2025 - 2026 Ping Identity Corporation. All rights reserved.
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */
import React, { useContext } from 'react';
import { ThemeContext } from '../../context/theme.context.js';

const Text = ({ collector, inputName, updater }) => {
  const theme = useContext(ThemeContext);

  return (
    <div className={`cstm_form-floating form-floating mb-3`}>
      <input
        className={`cstm_form-control form-control bg-transparent ${theme.textClass} ${theme.borderClass}`}
        id={inputName}
        name={inputName}
        onChange={(e) => updater(e.target.value)}
        type="text"
      />
      <label htmlFor={inputName}>{collector.output.label}</label>
    </div>
  );
};

export default Text;
